### ISSUE TITLE
Nested stack reference to InstanceProfile triggers E2502 Property IamInstanceProfile should relate to AWS::IAM::InstanceProfile

### ISSUE THREAD

**Issue by gene1wood** [awslabs/cfn-python-lint#408]
*cfn-lint version: `0.8.1`*

# Description of issue

When using nested stacks and passing IamInstanceProfile ARNs between stacks, E2502 is triggered though it shouldn't be.

# Steps to reproduce

Create a parent template like this
```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  IAMInstanceProfile:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: https://s3-us-west-2.amazonaws.com/example-bucket/example-instance-profile.yml
  Instance:
    Type: AWS::CloudFormation::Stack
    Properties:
      Parameters:
        IamInstanceProfile: !GetAtt IAMInstanceProfile.Outputs.InstanceProfileArn
      TemplateURL: https://s3-us-west-2.amazonaws.com/example-bucket/example-instance.yml
```
and a child template like this

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  InstanceProfile:
    Type: AWS::IAM::InstanceProfile
    Properties:
      Roles:
        - ExampleRole
Outputs:
  InstanceProfileArn:
    Value: !GetAtt InstanceProfile.Arn
```

# Expected results

The `IamInstanceProfile` parameter in the parent template's `Instance` sub-stack resource definition does indeed contain a valid IAM Instance Profile ARN (passed in from the `IAMInstanceProfile` sub-stack resource and as a result, there should be no error.

Ideally cfn-lint would recognize that `GetAtt` is referencing an output from another stack which very well could be an InstanceProfile ARN and as a result, optimistically not report this error.

Alternatively, if cfn-lint could introspect the sub-stack and determine the object type of the output, it would know whether or not it was the correct object type.

# Actual results

cfn-lint reports the error

> E2502 Property IamInstanceProfile should relate to AWS::IAM::InstanceProfile for Resources/Instance/Properties/Parameters/IamInstanceProfile/Fn::GetAtt
> example-parent.yml:11:9

**Comment by kddejong** [comment]
Got it.  I'll double check that.  If it relates to a nested stack we will just bail cause we can't verify anything at that point.
